operatorcourier/identify.py

```diff
@@ -1,16 +1,34 @@
-import yaml
+from yaml import safe_load
+from yaml import MarkedYAMLError
+import logging
+
+logger = logging.getLogger(__name__)
+
 
 def get_operator_artifact_type(operatorArtifactString):
-    """get_operator_artifact_type takes a yaml string and determines if it is one of the 
+    """get_operator_artifact_type takes a yaml string and determines if it is one of the
     expected bundle types: ClusterServiceVersion, CustomResourceDefinition, or Package.
 
     :param operatorArtifactString: Yaml string to type check
     """
-    operatorArtifact = yaml.safe_load(operatorArtifactString)
-    if isinstance(operatorArtifact, dict):
-        if "packageName" in operatorArtifact:
-            return "Package"
-        elif "kind" in operatorArtifact:
-            if operatorArtifact["kind"] in ("ClusterServiceVersion", "CustomResourceDefinition"):
-                return operatorArtifact["kind"]
-    raise ValueError('Courier requires valid CSV, CRD, and Package files')
+    try:
+        operatorArtifact = safe_load(operatorArtifactString)
+    except MarkedYAMLError:
+        logger.error("Courier requires valid input YAML files")
+        raise ValueError('Courier requires valid input YAML files')
+    else:
+        artifact_type, artifact_name = None, None
+        if operatorArtifact is not None:
+            if "packageName" in operatorArtifact:
+                artifact_type = "Package"
+                artifact_name = operatorArtifact['packageName']
+            elif "kind" in operatorArtifact:
+                if operatorArtifact["kind"] in ("ClusterServiceVersion", "CustomResourceDefinition"):
+                    artifact_type = operatorArtifact["kind"]
+                    artifact_name = operatorArtifact['metadata']['name']
+            if artifact_type is not None:
+                logger.info(f'Parsed {artifact_type}: {artifact_name}')
```

Comment:
No format strings (`f'...'`)! It is supported only in py 3.7+. We need older python versions too at least 3.6.

Also this logging is not lazy, please use `logger.info('Parsed %s: %s', artifact_type, artifact_name)` instead.

Reply:
Ah sorry format strings are supported in py 3.6, but there is still that lazy not lazy logging.